### PR TITLE
Record cases for Autorest 3.0.x

### PR DESCRIPTION
**Pull request by Youri970410**
## Description

Record cases for Autorest 3.0.x and fix custom issue.

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/CostManagement/test/New-AzCostManagementQueryComparisonExpressionObject.Tests.ps1 [hunk 1]
```diff
@@ -14,7 +14,7 @@ while(-not $mockingPath) {
 Describe 'New-AzCostManagementQueryComparisonExpressionObject' {
     It '__AllParameterSets' {
         { 
-            $orDimension = New-AzCostManagementQueryComparisonExpressionObject -Name 'ResourceLocation' -Value @('East US', 'West Europe')  
+            $orDimension = New-AzCostManagementQueryComparisonExpressionObject -Name 'ResourceLocation' -Operator 'In' -Value @('East US', 'West Europe')
```
Comment (wyunchi-ms) [inline src/CostManagement/test/New-AzCostManagementQueryComparisonExpressionObject.Tests.ps1:17]: Please remove `Operator`

File: src/CostManagement/test/New-AzCostManagementQueryFilterObject.Tests.ps1 [hunk 2]
```diff
@@ -14,11 +14,11 @@ while(-not $mockingPath) {
 Describe 'New-AzCostManagementQueryFilterObject' {
     It '__AllParameterSets' {
         { 
-            $orDimension = New-AzCostManagementQueryComparisonExpressionObject -Name 'ResourceLocation' -Value @('East US', 'West Europe') 
-            $orTag = New-AzCostManagementQueryComparisonExpressionObject -Name 'Environment' -Value @('UAT', 'Prod') 
+            $orDimension = New-AzCostManagementQueryComparisonExpressionObject -Name 'ResourceLocation' -Operator 'In' -Value @('East US', 'West Europe')
```
Comment (wyunchi-ms) [inline src/CostManagement/test/New-AzCostManagementQueryFilterObject.Tests.ps1:17]: Please remove `Operator`.

File: src/CostManagement/custom/Update-AzCostManagementExport.ps1 [hunk 3]
```diff
@@ -232,15 +232,73 @@ param(
...
+            $hasScheduleRecurrence = $PSBoundParameters.Remove('ScheduleRecurrence')
+            $hasScheduleStatus = $PSBoundParameters.Remove('ScheduleStatus')
+            $hasTimePeriodFrom = $PSBoundParameters.Remove('TimePeriodFrom')
+            $hasTimePeriodTo = $PSBoundParameters.Remove('TimePeriodTo')
             if($PSBoundParameters['InputObject'] -ne $null)
             {
-                $InputExportObject = $PSBoundParameters['InputObject']
-                $getExport = Get-AzCostManagementExport -InputObject $InputExportObject
+                # $InputExportObject = $PSBoundParameters['InputObject']
+                $getExport = Get-AzCostManagementExport @PSBoundParameters
```
Comment (wyunchi-ms) [inline src/CostManagement/custom/Update-AzCostManagementExport.ps1:253]: You need to create a new `BoundParameters` for `Get` but not reuse `PSBoundParameters`.

File: src/CostManagement/custom/Update-AzCostManagementExport.ps1 [hunk 4]
```diff
@@ -232,15 +232,73 @@ param(
     process {
         try {
             $getExport
```
Comment (wyunchi-ms) [inline src/CostManagement/custom/Update-AzCostManagementExport.ps1:234]: `$getExport` need to be removed.

File: src/CostManagement/custom/Update-AzCostManagementExport.ps1 (hunk 3 above)
Comment (Youri970410) [reply src/CostManagement/custom/Update-AzCostManagementExport.ps1]: Got it,thanks

File: src/CostManagement/test/New-AzCostManagementQueryComparisonExpressionObject.Tests.ps1 (hunk 1 above)
Comment (Youri970410) [reply src/CostManagement/test/New-AzCostManagementQueryComparisonExpressionObject.Tests.ps1:17]: It cannot be removed now.

File: src/CostManagement/test/New-AzCostManagementQueryFilterObject.Tests.ps1 (hunk 2 above)
Comment (Youri970410) [reply src/CostManagement/test/New-AzCostManagementQueryFilterObject.Tests.ps1:17]: Thanks,but It cannot be removed now.

**Review by wyunchi-ms — APPROVED** [review]

File: src/CostManagement/custom/Update-AzCostManagementExport.ps1 [hunk 5]
```diff
@@ -231,16 +231,68 @@ param(
...
+                $null = $PSBoundParameters.Add("ScheduleRecurrence",$ScheduleRecurrence)
+            }
+            if ($hasScheduleStatus){
+                $null = $PSBoundParameters.Add("ScheduleStatus",$ScheduleStatus)
+            }
+            if ($hasTimePeriodFrom){
+                $null = $PSBoundParameters.Add("TimePeriodFrom",$TimePeriodFrom)
+            }
+            if ($hasTimePeriodTo){
+                $null = $PSBoundParameters.Add("TimePeriodTo",$TimePeriodTo)
```
Comment (wyunchi-ms) [inline src/CostManagement/custom/Update-AzCostManagementExport.ps1:293]: need to remove these lines

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**